config/tests/samples/create/samples.go

```diff
@@ -207,12 +216,13 @@ func DeleteResources(t *Harness, unstructs []*unstructured.Unstructured) {
 	var wg sync.WaitGroup
```

Comment:
In the SoftDeleteCase, I think the deletion timestamp will always be set, should we just skip `waitForDeleteToComplete`?  (This also motivates the potentially simpler name `WaitForDelete: false`)